src/go/rpk/pkg/cli/cluster/partitions/transfer_leader.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newTransferLeaderCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var partitionArg string
+
+	cmd := &cobra.Command{
+		Use:   "transfer-leadership",
```

Comment:
Wonder if the command name should match the filename

Reply:
Yes, we tend to match file name with the command name, but it's not strict (multiple commands in one file, easier to find, etc...)

For example:
- https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/topic/config.go (config -> alter-config).

Reply:
I misunderstood Paul's question. Originally I thought `-` vs `_`, but Paul's point was `leader` vs `leadership`. I agree with it and make a change to the file name.